### PR TITLE
feature/cplp 1598 refactor approle

### PR DESCRIPTION
**Pull request by qxz2mqe**
https://jira.catena-x.net/browse/CPLP-1598

### CODE REVIEW

File: src/marketplace/Apps.Service/Controllers/AppReleaseProcessController.cs [hunk 1]
```diff
@@ -107,14 +107,12 @@ public async Task<IActionResult> UpdateApp([FromRoute] Guid appId, [FromBody] Ap
     [HttpPost]
     [Route("{appId}/role")]
     [Authorize(Roles = "edit_apps")]
-    [ProducesResponseType(StatusCodes.Status204NoContent)]
+    [ProducesResponseType(typeof(IEnumerable<AppRoleData>), StatusCodes.Status200OK)]
```
Comment (Phil91) [inline src/marketplace/Apps.Service/Controllers/AppReleaseProcessController.cs:110]: Please add a <response code="200" comment for swagger

File: src/marketplace/Apps.Service/BusinessLogic/AppReleaseBusinessLogic.cs [hunk 2]
```diff
@@ -215,24 +215,26 @@ public Task AddAppUserRoleAsync(Guid appId, IEnumerable<AppUserRole> appAssigned
...
+    private async Task<IEnumerable<AppRoleData>> InsertAppUserRoleAsync(Guid appId, IEnumerable<AppUserRole> appAssignedDesc, string userId)
     {
         var appReleaseRepository = _portalRepositories.GetInstance<IAppReleaseRepository>();
 
         if (!await appReleaseRepository.IsProviderCompanyUserAsync(appId, userId).ConfigureAwait(false))
         {
             throw new NotFoundException($"Cannot identify companyId or appId : User CompanyId is not associated with the same company as AppCompanyId");
         }
-
+        var roelData = new List<AppRoleData>();
```
Comment (Phil91) [inline src/marketplace/Apps.Service/BusinessLogic/AppReleaseBusinessLogic.cs:226]: typo. please adjust to roleData

**Review by Phil91 — CHANGES_REQUESTED** [review]
Please write some UnitTests for the changed code.

File: src/marketplace/Apps.Service/BusinessLogic/AppReleaseBusinessLogic.cs [hunk 3]
```diff
@@ -215,24 +215,26 @@ public Task AddAppUserRoleAsync(Guid appId, IEnumerable<AppUserRole> appAssigned
...
         {
             var appRole = appReleaseRepository.CreateAppUserRole(appId, indexItem.role);
+            roelData.Add(new AppRoleData(appRole.Id, indexItem.role));
             foreach (var item in indexItem.descriptions)
             {
                 appReleaseRepository.CreateAppUserRoleDescription(appRole.Id, item.languageCode, item.description);
             }
         }
         await _portalRepositories.SaveAsync().ConfigureAwait(false);
+        return roelData.AsEnumerable();
```
Comment (ntruchsess) [inline src/marketplace/Apps.Service/BusinessLogic/AppReleaseBusinessLogic.cs:237]: List implements IEnumerable, the call to AsEnumerable() is redundant. Please remove

**Review by ntruchsess — CHANGES_REQUESTED** [review]

File: src/marketplace/Apps.Service/Controllers/AppReleaseProcessController.cs [hunk 4]
```diff
@@ -104,10 +104,11 @@ public async Task<IActionResult> UpdateApp([FromRoute] Guid appId, [FromBody] Ap
     /// <remarks>Example: POST: /api/apps/appreleaseprocess/{appId}/role</remarks>
     /// <response code="400">If sub claim is empty/invalid or user does not exist, or any other parameters are invalid.</response>
     /// <response code="404">App does not exist.</response>
+    /// <response code="201">Return the created role Id and role name.</response>
     [HttpPost]
     [Route("{appId}/role")]
     [Authorize(Roles = "edit_apps")]
-    [ProducesResponseType(typeof(IEnumerable<AppRoleData>), StatusCodes.Status200OK)]
+    [ProducesResponseType(typeof(IEnumerable<AppRoleData>), StatusCodes.Status201Created)]
```
Comment (Phil91) [inline src/marketplace/Apps.Service/Controllers/AppReleaseProcessController.cs:111]: If you want a 201 you need to change the implementation to return a `CreatedAtRoute`

**Review by Phil91 — CHANGES_REQUESTED** [review]

File: src/marketplace/Apps.Service/BusinessLogic/AppReleaseBusinessLogic.cs (hunk 3 above)
Comment (qxz2mqe) [reply src/marketplace/Apps.Service/BusinessLogic/AppReleaseBusinessLogic.cs]: done

File: src/marketplace/Apps.Service/Controllers/AppReleaseProcessController.cs (hunk 4 above)
Comment (qxz2mqe) [reply src/marketplace/Apps.Service/Controllers/AppReleaseProcessController.cs]: fixed

**Comment by qxz2mqe** [comment]
Please write some UnitTests for the changed code--Added

**Review by ntruchsess — APPROVED** [review]

File: src/portalbackend/PortalBackend.DBAccess/Repositories/AppReleaseRepository.cs [hunk 5]
```diff
@@ -49,33 +49,4 @@ public  Task<Guid> GetCompanyUserIdForOfferUntrackedAsync(Guid offerId, string u
             .Select(x=>x.ProviderCompany!.CompanyUsers.First(companyUser => companyUser.IamUser!.UserEntityId == userId).Id)
```
Comment (Phil91) [inline src/portalbackend/PortalBackend.DBAccess/Repositories/AppReleaseRepository.cs:49]: Would it make sense to move this method into the offer repository as well? than we could get rid of the AppReleaseRepository.

Comment (ntruchsess) [reply src/portalbackend/PortalBackend.DBAccess/Repositories/AppReleaseRepository.cs]: done

**Review by Phil91 — APPROVED** [review]

**Pull request merged**